### PR TITLE
Remove almost all manual memory management from gr-blocks

### PR DESCRIPTION
**Pull request by ThomasHabets**
The remaining one is `tcp_server_sink`, which is a bit trickier, so leaving it for a future PR.

### CODE REVIEW

File: gr-blocks/lib/udp_source_impl.cc [hunk 1]
```diff
@@ -105,7 +101,7 @@ void udp_source_impl::disconnect()
     d_udp_thread.join();
 
     d_socket->close();
-    delete d_socket;
+    d_socket = nullptr;
```
Comment (mbr0wn) [inline gr-blocks/lib/udp_source_impl.cc]: Don't you usually do `d_socket.reset()`?

Comment (ThomasHabets) [reply gr-blocks/lib/udp_source_impl.cc]: [Looks to me like `reset()` mandates an argument until C++17](https://en.cppreference.com/w/cpp/memory/unique_ptr/reset), so could also be `d_socket.reset(nullptr);`.

I have no preference. Whatever you prefer.

Comment (rear1019) [reply gr-blocks/lib/udp_source_impl.cc]: ``reset()`` has a default argument, both in pre and post C++17. The things which differ in pre/post C++17 only apply to array types (``T[]``), not to the “scalar” case here.

As far is the freeing of the memory is concerned: I prefer calling ``reset()`` to state the intent more clearly. Apart from that: LGTM.

Comment (mbr0wn) [reply gr-blocks/lib/udp_source_impl.cc]: I have no strong opinion, but @rear1019's sounds good.

Comment (ThomasHabets) [reply gr-blocks/lib/udp_source_impl.cc]: Fixed. Here and in udp sink.

**Pull request merged**